### PR TITLE
Swap ordering of thread configuration in Sycl

### PR DESCRIPTION
**Pull request by artv3**
Sycl uses the outermost index as the fastest, this change enables consistency when using RAJA::launch with other device back ends.

For reference this page has a good description on thread indexing using sycl: 
https://www.intel.com/content/www/us/en/docs/dpcpp-compatibility-tool/developer-guide-reference/2023-2/cuda-and-sycl-programming-model-comparison.html

See the Thread Indexing section.

### CODE REVIEW

**Review by rchen20 — APPROVED** [review]

**Review by johnbowen42 — APPROVED** [review]

**Review by long58 — APPROVED** [review]

**Comment by artv3** [comment]
@rhornung67 , did you want to take a look at this PR? Or should I just merge?

**Review by rhornung67 — COMMENTED** [review]
Please add some notes about this in the user guide. Also, did you notice any performance differences with this change?

**Comment by artv3** [comment]
> Please add some notes about this in the user guide. Also, did you notice any performance differences with this change?

@rhornung67 I just pushed up a note on the docs. Please let me know if any adjustments are needed. I held off on commenting about kernel as we need to double check and we can follow up after. I also don't have a sense on performance changes as I had not looked at timings and the kernels in the raja perf suite were generating incorrect answers prior to this change.

File: docs/sphinx/user_guide/feature/policies.rst [hunk 1]
```diff
@@ -524,10 +524,26 @@ write more explicit policies.
...
+	  dimension 0 has the longest stride. This is
+	  important to note as the ordering is reverse
+	  compared to the CUDA and HIP programming models.
+	  CUDA and HIP employ a x/y/z ordering in which
+	  dimension x has the unit stride.
+
+	  When using RAJA launch thread and team configuration
+	  follows CUDA and HIP programming models and is always
+	  configured in three-dimensions. This means that dimension
+	  2 always exist and should be used as one would the
```
Comment (rhornung67) [inline docs/sphinx/user_guide/feature/policies.rst:544]:
```suggestion
	  2 always exists and should be used as one would the
```

**Review by rhornung67 — COMMENTED** [review]
Minor doc changes.

**Review by rhornung67 — APPROVED** [review]

**Comment by rhornung67** [comment]
@artv3 if you don't want to keep the branch, please delete it.

**Pull request merged**